cookbook/grid_pad.py

```diff
@@ -0,0 +1,79 @@
+"""
+Gridding: Pad gridded data
+"""
+
+from fatiando import mesher,gridder
+from fatiando.gravmag import prism
+from fatiando.vis import mpl
+import numpy as np
+
+# Generate gridded data
+shape = (101,172)
+x,y,z = gridder.regular((-5000, 5000, -5000, 5000), shape, z=-150)
+model = [mesher.Prism(-4000, -3000, -4000, -3000, 0, 2000, {'density':1000}),
+        mesher.Prism(-1000, 1000, -1000, 1000, 0, 2000, {'density':-900}),
+        mesher.Prism(2000, 4000, 3000, 4000, 0, 2000, {'density':1300})]
+gz = prism.gz(x, y, z, model)
+gz = gz.reshape(shape)
+
+# Pad arrays with all the padding options
+pads = []
+xy = np.zeros((len(x),2))
+xy[:,0] = x[:]
+xy[:,1] = y[:]
```

Comment:
@drandykass the handling of the coordinate arrays seems a bit too complicated. Wouldn't it be better to pass along (and return) a list of coordinate arrays? That would also avoid the memory copying involved in creating a new 2D array (the list holds only references to the arrays, not their values).